### PR TITLE
File flaky issue with limited number of successful builds

### PR DESCRIPTION
**Pull request by keyonghan**
This solves: https://github.com/flutter/flutter/issues/93000.

Added a test for the proposed change, but test validation is blocked on https://github.com/flutter/flutter/issues/90139.
Landing the change to unblock the auto bot, and the test needs validation when migrating NNBD.

### CODE REVIEW

File: app_dart/lib/src/request_handlers/flaky_handler_utils.dart [hunk 1]
```diff
@@ -77,7 +79,7 @@ Flaky builds:
 ${_issueBuildLinks(builder: statistic.name, builds: statistic.flakyBuilds!)}
 
 Succeeded builds (3 most recent):
-${_issueBuildLinks(builder: statistic.name, builds: statistic.succeededBuilds!.sublist(0, 3))}
+${_issueBuildLinks(builder: statistic.name, builds: statistic.succeededBuilds!.sublist(0, statistic.succeededBuilds!.length >= kSuccessBuildNumberLimit ? kSuccessBuildNumberLimit : statistic.succeededBuilds!.length))}
```
Comment (CaseyHillers) [inline app_dart/lib/src/request_handlers/flaky_handler_utils.dart:82]: This is hard to follow. Can we assign this to a variable and pass it instead?

File: app_dart/test/request_handlers/file_flaky_issue_and_pr_test_data.dart [hunk 2]
```diff
@@ -203,6 +203,39 @@ const String expectedSemanticsIntegrationTestPullRequestBody = '''
...
+    flakyRate: 0.5,
+    flakyBuilds: <String>['103', '102', '101'],
+    succeededBuilds: <String>['201'],
+    recentCommit: 'abc',
+    flakyBuildOfRecentCommit: '103',
+  )
+];
+
+const String expectedLimitedNumberOfBuildsResponseBody = '''
+<!-- meta-tags: To be used by the automation script only, DO NOT MODIFY.
```
Comment (CaseyHillers) [inline app_dart/test/request_handlers/file_flaky_issue_and_pr_test_data.dart:218]: TIL! This is neat

File: app_dart/lib/src/request_handlers/flaky_handler_utils.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/lib/src/request_handlers/flaky_handler_utils.dart]: Updated.

**Review by CaseyHillers — APPROVED** [review]
